### PR TITLE
#294: restore default filenames when a tag is missing

### PR DESCRIPTION
**Pull request by AdrienPoupa**
@soncaokim asking for your review here since Discog is your baby :)

Basically I am just adding a fallback when there is no title/artist/album.

Let me know if that's OK.

Also, how can we fix it for existing users? ie: how to force the discog cache to clear so they can see the filenames again?

### CODE REVIEW

**Comment by soncaokim** [comment]
> Basically I am just adding a fallback when there is no title/artist/album.

Thanks Adrien.
I'd add as well the fallback for empty genre as well. 

> Also, how can we fix it for existing users? ie: how to force the discog cache to clear so they can see the filenames again?

I'd bump the Discog DB version (from 2 to 3) to force clearing the DB. In the current master implementation, this will start off showing users with MediaStore info, and gradually filling/replacing the Discog with ID3 tags.

**Comment by soncaokim** [comment]
@AdrienPoupa I can have a look at this later today (for the comments you've left). And update directly this PR.
-> Tell me :-)

**Comment by AdrienPoupa** [comment]
Thanks for the input! I increased the DB version.

If this looks good to you, we can merge and release.

**Comment by AdrienPoupa** [comment]
About the empty Genre, surprisingly it's not handled by Discography. I am guessing this is part of your v2 PR? If yes we can delay it as it's not as urgent

**Comment by soncaokim** [comment]
> About the empty Genre, surprisingly it's not handled by Discography. I am guessing this is part of your v2 PR? If yes we can delay it as it's not as urgent

Yes I will tackle this empty genre on v2.

**Review by soncaokim — APPROVED** [review]

File: app/src/main/java/com/poupa/vinylmusicplayer/provider/Discography.java [hunk 1]
```diff
@@ -199,6 +201,36 @@ private void extractTags(@NonNull Song song) {
         }
     }
 
+    private String getArtistName(Tag tags) {
+        String artist = tags.getFirst(FieldKey.TITLE);
```
Comment (soncaokim) [inline app/src/main/java/com/poupa/vinylmusicplayer/provider/Discography.java:205]: Hmm, shouldnt we get FieldKey.ARTIST instead here?

**Pull request merged**